clients/java/zms/src/main/java/com/yahoo/athenz/zms/ZMSClient.java

```diff
@@ -322,15 +317,42 @@ String lookupZMSUrl() {
             AthenzConfig conf = JSON.fromBytes(Files.readAllBytes(path), AthenzConfig.class);
             url = conf.getZmsUrl();
         } catch (Exception ex) {
-            LOGGER.error("Unable to extract ZMS Url from {} exc: {}",
-                    confFileName, ex.getMessage());
+            LOGGER.error("Unable to extract ZMS Url from {} exc: {}", confFileName, ex.getMessage());
         }
 
         return url;
     }
 
-    ClientBuilder getClientBuilder() {
-        return ClientBuilder.newBuilder();
+    protected PoolingHttpClientConnectionManager createConnectionPooling(SSLContext sslContext) {
+        if (sslContext == null) {
+            return null;
+        }
+        Registry<ConnectionSocketFactory> registry = RegistryBuilder.<ConnectionSocketFactory>create()
+                .register("https", new SSLConnectionSocketFactory(sslContext))
+                .register("http", new PlainConnectionSocketFactory())
+                .build();
+        PoolingHttpClientConnectionManager poolingHttpClientConnectionManager
+                = new PoolingHttpClientConnectionManager(registry, dnsResolver);
+
+        //route is host + port.  Since we have only one, set the max and the route the same
```

Comment:
Not sure about the comment. It says "set the max and the route the same" but then default max per route is 30 while max total is 20